packages/spacecat-shared-google-client/src/index.js

```diff
@@ -32,6 +32,7 @@ export default class GoogleClient {
       config.clientSecret,
       config.redirectUri,
     );
+    this.apiKey = config.apiKey;
```

Comment:
This may warrant a discussion is this is appropriate way to auth with these APIs or not.

My understanding is that the oath client used for the GSC apis that exist in this module uses OAUTH on behalf of the customer. In this case, I don't think we want to do that. Both of these APIs support API key auth, subject to some rate limits. Would request us to create and manage that Key, but that I assume should be simple enough. But happy to discuss and if using auth is the preferred route, we can do that.

https://developers.google.com/speed/docs/insights/v5/get-started#APIKey
https://developer.chrome.com/docs/crux/api#crux_api_key